### PR TITLE
Change background color and text color to improve text readability for visually impaired prompt.

### LINKED ISSUE #732: "Describe the photo" caption is invisible on some images
The text saying "Describe the photo" is not visible on certain kinds of images.

Maybe "Describe the photo" could be pale text on a dark transparent background so that it is visible on all images? Or alternatively, pale text with thick dark outlining?

See an example below:

descripexample

### PR DESCRIPTION
**Pull request by ivannnito**
First PR to this project. Tested these changes and text can be read with any image more clearly. Please let me know if you have any questions.

Closes #732

### CODE REVIEW

**Comment by j-f1** [comment]
Welcome to the project! I ran your PR locally and I felt like the dark border was a little too tightly wrapped around the text input. Do you think it would be possible to make it extend to the edge of the image on the left/right/bottom sides and an equivalent amount up on the top side so that there’s a bit of padding there? Otherwise I think it’s a great improvement :)

**Comment by ivannnito** [comment]
Hey Jed! Thank you so much for such a warm welcome. I'll get to work on it soon now that I'm back on my computer. Thanks for the suggestion, I'll let you know if the change is possible and will update if it is. Thanks again!

**Comment by ivannnito** [comment]
Hey @j-f1, I just pushed the updated version of the prompt, let me know what you think of it!

**Comment by ivannnito** [comment]
Accidentally closed.. Wanted to let you know that I just updated the prompt @j-f1, let me know what you think of the changes!

File: MastodonSDK/Sources/MastodonUI/Scene/ComposeContent/Attachment/AttachmentView.swift [hunk 1]
```diff
@@ -46,12 +46,16 @@ public struct AttachmentView: View {
...
-                                    .foregroundColor(.white)
-                                    .placeholder(placeholder, when: viewModel.caption.isEmpty)
-                                    .padding(8)
+                                    .foregroundColor(.white).padding(EdgeInsets(top: 0, leading: 8, bottom: 0, trailing: 4))
+                                    .background(Color.black.opacity(0.4))
+                                    .placeholder(when: viewModel.caption.isEmpty) {
+                                        Text(placeholder).foregroundColor(.white).padding(EdgeInsets(top: 0, leading: 8, bottom: 0, trailing: 4))
+                                            .lineLimit(1)
+                                    }
+                                    .padding(EdgeInsets(top: 6, leading: 0, bottom: 10, trailing: 0))
```
Comment (zeitschlag) [inline MastodonSDK/Sources/MastodonUI/Scene/ComposeContent/Attachment/AttachmentView.swift]: First of all: Thank you for your work 🥳 

When I run this, it's slightly better than before, but still difficult to see:



Do you know, what I mean? What do you think about this instead?



The color-values match (well, at least, they should) with what @j-f1 built for alt-text.

```swift
TextField("", text: $viewModel.caption)
    .lineLimit(1)
    .textFieldStyle(.plain)
    .padding(8)
    .placeholder(when: viewModel.caption.isEmpty) {
        Text(placeholder)
            .padding(8)
            .lineLimit(1)
    }
    .background(Color.black.opacity(0.85))
    .foregroundColor(.white)
    .padding(8)
```

One could throw in some rounded corners, too, but I'd leave that up to you :)

**Comment by j-f1** [comment]
Maybe something like this?

- keeps consistency with how alt text is displayed in app
- separate UI encourages writing longer alt text
- could possibly run Live Text analysis and put any detected text in (or add a button for this?)

the one small cheat I’ve done here is reduce the default height of the main input field so that the +ALT button is visible (since currently the whole control is covered by the keyboard on my phone). Maybe that’s a reason to move the button to the top-left? or somehow encourage users to scroll down? or make the images smaller?

Screenshot_2022-12-26 12 12 42

**Comment by Gargron** [comment]
Note, the design on Figma had a black gradient on the bottom of the thumbnail (same as on the web app) -- this was missed during implementation.

**Comment by zeitschlag** [comment]
Ah, you're right, I should've looked there 🙈
This is how it looks there:

Compose - iPhone - Uploaded

- Linear Gradient (black, from alpha 0 to alpha 1)
- height in screenshot: 80pt

**Comment by j-f1** [comment]
Also here's a sample alt text to use that's pretty close to the 1000 char limit, for perspective:

<details>

> A picture of a big fluffy white cat sitting on a grey sofa. The cat has white fur with black spots all over its body, and it is looking at the camera with its big blue eyes. The cat is sitting with its back legs on the sofa and its front legs in the air. The background of the image is a white wall.

> This is a picture of a big fluffy white cat with black spots, sitting on a grey sofa. The cat has its back legs on the sofa and its front legs in the air, and is looking at the camera with its big blue eyes. The background of the image is a white wall. This cat looks content and comfortable, and is a great example of a friendly and loving feline. This image conveys a feeling of love and contentment, as well as coziness and comfort.

> The cat's bright blue eyes make it seem like it is looking directly at the viewer, and it seems almost like it is asking for a cuddle. The grey sofa adds to the cozy atmosphere, and the white background of the image helps to draw attention to the cat. Overall, this image provides a warm and inviting feeling.

> This image is great for anyone who is a fan of cats, and it's perfect for bringing a bit of warmth and comfort into any room. Whether it's used as a wallpaper or an art print, this image is sure to bring a smile to anyone's face. Its bright blue eyes make it seem like it's looking directly at the viewer, and its warm colors create a cozy atmosphere.

</details>

File: MastodonSDK/Sources/MastodonUI/Scene/ComposeContent/Attachment/AttachmentView.swift (hunk 1 above)
Comment (zeitschlag) [reply MastodonSDK/Sources/MastodonUI/Scene/ComposeContent/Attachment/AttachmentView.swift]: Kind of obsolete now, as there should be a gradient?

**Review by kimar — APPROVED** [review]
Thank you @ivannnito ! I've slightly tweaked the appearance to match the Figma design. Thanks a lot for your contribution!

**Comment by j-f1** [comment]
It looks like the input field only allows one line of alt text, which can make it harder to read descriptions of e.g. long passages with multiple paragraphs.

**Comment by kimar** [comment]
@j-f1 yes that's currently true. I'm going to merge those changes now and add a task to reconsider multi-line alt input.

**Comment by wayneca** [comment]
(after publishing this I saw jf-1 was included on this one - so what I asked about would have been considered.)

Another related issue was that when only some of the image was shown, the alt text field was on part of the image that was not in view at all.  If you used touch to scroll to see it, then when you released that to type it jumped back so you could not see it again.  So, no way to see it, select it and type at the same time.  This was on an ipad, either with or without an attached keyboard.  Is that fixed with this change too?

**Comment by wayneca** [comment]
#813     @danyowdee  point 1 is the same as  what I was referring to above.

**Comment by danyowdee** [comment]
> Maybe something like this?
> 
> * keeps consistency with how alt text is displayed in app
> * separate UI encourages writing longer alt text
> * could possibly run Live Text analysis and put any detected text in (or add a button for this?)
> 
> the one small cheat I’ve done here is reduce the default height of the main input field so that the +ALT button is visible (since currently the whole control is covered by the keyboard on my phone). Maybe that’s a reason to move the button to the top-left? or somehow encourage users to scroll down? or make the images smaller?
> 
> Screenshot_2022-12-26 12 12 42

@j-f1 this is pretty much what I had in mind when I wrote #813.

**Comment by ivannnito** [comment]
> Thank you @ivannnito ! I've slightly tweaked the appearance to match the Figma design. Thanks a lot for your contribution!

Thank you so much! Sorry for my delayed reply as I have been moving recently..

**Pull request merged**